File: mobile-app/app/src/main/java/com/margwatch/data/model/Complaint.kt
package com.margwatch.data.model

data class Complaint(
    val id: String,
    val title: String,
    val description: String,
    val category: String, // IssueCategory enum from backend
    val status: String, // ComplaintStatus enum from backend
    val latitude: Float,
    val longitude: Float,
    val address: String?,
    val imageUrl: String?, // JSON array of image URLs from Cloudinary
    val imageUrls: List<String> = emptyList(), // Parsed image URLs for easier use
    val imageCount: Int = 1,
    // ML Prediction Fields
    val mlCategory: String?, // ML predicted category
    val mlConfidence: Float?, // ML confidence score (0-1)
    val mlModelVersion: String?, // ML model version used
    val mlProcessingTime: Float?, // ML processing time in seconds
    val userId: String,
    val user: User? = null, // Include user details if needed
    val approvedBy: String? = null,
    val approvedAt: String? = null,
    val createdAt: String,
    val updatedAt: String
)

File: packages/shared-types/src/interfaces/Complaint.ts
import { ComplaintStatus, IssueCategory } from '../enums';
import { User } from './User';
import { WorkOrder } from './WorkOrder';

/**
 * Complaint interface
 * Represents a complaint in the MargWatch system
 */
export interface Complaint {
  id: string;
  title: string;
  description: string;
  category: IssueCategory;
  status: ComplaintStatus;
  latitude: number;
  longitude: number;
  address?: string;
  imageUrl?: string; // JSON string of image URLs
  imageCount?: number;
  mlCategory?: IssueCategory;
  mlConfidence?: number;
  mlModelVersion?: string;
  mlProcessingTime?: number;
  userId: string;
  user: {
    id: string;
    firstName: string;
    lastName: string;
    email: string;
  };
  approvedBy?: string;
  approvedAt?: Date | string;
  createdAt: Date | string;
  updatedAt: Date | string;
  workOrders?: WorkOrder[];
  updates?: ComplaintUpdate[];
}

/**
 * Complaint creation request interface
 */
export interface CreateComplaintRequest {
  title: string;
  description: string;
  latitude: number;
  longitude: number;
  address?: string;
  category?: IssueCategory;
}

/**
 * Complaint update request interface
 */
export interface UpdateComplaintRequest {
  status?: ComplaintStatus;
  description?: string;
  title?: string;
}

/**
 * Complaint filters interface
 */
export interface ComplaintFilters {
  page?: number;
  limit?: number;
  status?: ComplaintStatus | string;
  category?: IssueCategory | string;
  userId?: string;
  workerId?: string;
  search?: string;
}

/**
 * Complaint update interface
 */
export interface ComplaintUpdate {
  id: string;
  complaintId: string;
  status: ComplaintStatus | string;
  description?: string;
  imageUrl?: string;
  createdAt: Date | string;
}

/**
 * Heat map data interface
 */
export interface HeatMapData {
  latitude: number;
  longitude: number;
  count: number;
  category: IssueCategory;
}
